www/docs/getting-started/client.md

```diff
@@ -34,24 +34,51 @@ You can use the [session callback](/configuration/callbacks#session-callback) to
 
 The `useSession()` React Hook in the NextAuth.js client is the easiest way to check if someone is signed in.
 
-It works best when the [`<SessionProvider>`](#sessionprovider) is added to `pages/_app.js`.
+Make sure that [`<SessionProvider>`](#sessionprovider) is added to `pages/_app.js`.
 
 #### Example
 
 ```jsx
 import { useSession } from "next-auth/react"
 
 export default function Component() {
-  const [session, loading] = useSession()
+  const { data: session, status } = useSession()
 
-  if (session) {
+  if (status === "authenticated") {
     return <p>Signed in as {session.user.email}</p>
   }
 
   return <a href="/api/auth/signin">Sign in</a>
 }
 ```
 
+### Require session
+
+Due to the way how Next.js handles `getServerSideProps` and `getInitialProps`, every protected page load has to make a server-side request to check if the session is valid and then generate the requested page (SSR). This increases server load, and if you are good with making the requests from the client, there is an alternative. You can use `useSession` in a way that makes sure you always have a valid session. If after the initial loading state there was no session found, you can define the appropriate action to respond.
+
+The default behavior is to redirect the user to the sign-in page, from where - after a successful login - they will be sent back to the page they started on. You can also define an `action()` callback, if you would like to do something else.
+
```

Comment:
We must document here the different values of `status` for non TS users 💅🏽

Reply:
Agree! Good catch.